jib-core/src/test/java/com/google/cloud/tools/jib/tar/TarStreamBuilderTest.java

```diff
@@ -130,17 +127,40 @@ public void testToBlob_multiByte() throws IOException {
     Assert.assertEquals("crepecake", headerFile.getName());
     Assert.assertEquals(
         "asdf", new String(ByteStreams.toByteArray(tarArchiveInputStream), StandardCharsets.UTF_8));
-    Assert.assertEquals(timeFromTarArchiveEntry, headerFile.getModTime().toInstant());
 
     headerFile = tarArchiveInputStream.getNextTarEntry();
     Assert.assertEquals("jib", headerFile.getName());
     Assert.assertEquals(
         "jib", new String(ByteStreams.toByteArray(tarArchiveInputStream), StandardCharsets.UTF_8));
-    Assert.assertEquals(timeFromTarArchiveEntry, headerFile.getModTime().toInstant());
 
     Assert.assertNull(tarArchiveInputStream.getNextTarEntry());
   }
 
+  @Test
+  public void testToBlob_modificationTime() throws IOException {
+    testTarStreamBuilder.addByteEntry(
+        "foo".getBytes(StandardCharsets.UTF_8), "foo", Instant.ofEpochSecond(1234));
+    testTarStreamBuilder.addBlobEntry(
+        Blobs.from("bar"),
+        "bar".getBytes(StandardCharsets.UTF_8).length,
+        "bar",
+        Instant.ofEpochSecond(3));
+
+    ByteArrayOutputStream outStream = new ByteArrayOutputStream();
+    testTarStreamBuilder.writeAsTarArchiveTo(outStream);
+
+    TarArchiveInputStream tarInStream =
+        new TarArchiveInputStream(new ByteArrayInputStream(outStream.toByteArray()));
+
+    TarArchiveEntry headerFile = tarInStream.getNextTarEntry();
+    assertThat(headerFile.getName()).isEqualTo("foo");
+    assertThat(headerFile.getModTime().toInstant()).isEqualTo(Instant.ofEpochSecond(1234));
+
+    headerFile = tarInStream.getNextTarEntry();
```

Comment:
let's initialize all the `TarArchiveEntry headerFile` variables in the beginning? To keep the "arrange" and  "assert" blocks separate.